### PR TITLE
add unit test step in gradle ci

### LINKED ISSUE #11513: Gradle CI doesn't launch unit tests
It seems that the gradle ci don't launch unit test only IT.

I have tried to put a failing unit test and the ci was green 

Is this the behaviour or a misconfiguration on my side ?

A possible working solution

**github-actions.yml**

```
jobs:
  tests:
    name: tests
    runs-on: ubuntu-latest
    timeout-minutes: 20
    steps:
      - name: 'Setup: checkout project'
        uses: actions/checkout@v4
      - name: 'Setup: environment'
        id: setup
        uses: ./.github/actions/setup
      - name: Setup Gradle
        uses: gradle/gradle-build-action@v2
      - name: 'Test: run unit tests'
        run: ./gradlew clean test --no-daemon
      - name: 'Test: run integration tests'
        run: ./gradlew clean integrationTest --no-daemon
```

### PR DESCRIPTION
**Pull request by fabienpuissant**
Fix #11513

### CODE REVIEW

File: src/main/resources/generator/ci/github/actions/.github/workflows/github-actions-gradle.yml.mustache [hunk 1]
```diff
@@ -25,5 +25,7 @@ jobs:
         uses: ./.github/actions/setup
       - name: Setup Gradle
         uses: gradle/gradle-build-action@v2
+      - name: 'Test: run unit tests'
+        run: ./gradlew clean test --no-daemon
       - name: 'Test: run tests'
         run: ./gradlew clean integrationTest --no-daemon
```
Comment (renanfranca) [inline src/main/resources/generator/ci/github/actions/.github/workflows/github-actions-gradle.yml.mustache:31]: @fabienpuissant : As @atomfrede said before,  i think the clean should be removed from that command.

Thanks for fixing that issue!

Comment (fabienpuissant) [reply src/main/resources/generator/ci/github/actions/.github/workflows/github-actions-gradle.yml.mustache:31]: absolutely I've done this on my personal repo but not there xD

**Review by renanfranca — DISMISSED** [review]
Thank you for your work!

File: src/main/resources/generator/ci/github/actions/.github/workflows/github-actions-gradle.yml.mustache [hunk 2]
```diff
@@ -25,5 +25,7 @@ jobs:
         uses: ./.github/actions/setup
       - name: Setup Gradle
         uses: gradle/gradle-build-action@v2
+      - name: 'Test: run unit tests'
+        run: ./gradlew clean test --no-daemon
       - name: 'Test: run tests'
```
Comment (renanfranca) [inline src/main/resources/generator/ci/github/actions/.github/workflows/github-actions-gradle.yml.mustache:30]:
```suggestion
      - name: 'Test: run integration tests'
```

@fabienpuissant : I only saw that now, sorry.

**Pull request merged**